list/directory/list.js

```diff
@@ -0,0 +1,439 @@
+
+var INPUT_EVENT = 'input';
+
+var directoryListMixin = {
+    /**
+     * Component display name.
+     */
+    displayName: 'directory-list',
+
+    /**
+     * Mixins used by component.
+     * @type {Array}
+     */
+    mixins: [memoryMixin, stylable],
+
+    /**
+     * Sorted data.
+     * @type {Array}
+     */
+    sortedData: [],
+
+    /**
+     * search text.
+     * @type {string}
+     */
+    searchText: '',
+
+    /**
+     * search dom node.
+     * @type {object}
+     */
+    quickSearchDOMNode: undefined,
+
+    /** @inheritedDoc */
+    getDefaultProps() {
+        return {
+            /**
+             * Property name to be filtered by letters-filter component.
+             * @type {string}
+             */
+            fieldName: '',
+
+            /**
+             * Show all data.
+             * If true, list will display all data separated by letters.
+             * Otherwise, only data responding to filter criteria will be shown in the list.
+             * @type {boolean}
+             */
+            showAll: true,
+
+            /**
+             * Display group size.
+             * If true, display size of each group.
+             * @type {boolean}
+             */
+            displayGroupSize: false,
+
+            /**
+             * Data to be displayed in the list.
+             * @type {Array}
+             */
+            data: [],
+
+            /**
+             * Define an external quick search input.
+             * In this case, the default quick search will be omitted.
+             * @type {string}
+             */
+            quickSearchInput: undefined,
+
+            /**
+             * Add a default quick search input.
+             * @type {boolean}
+             */
+            withQuickSearch: true,
+
+            /**
+             * Search in all fields.
+             * @type {boolean}
+             */
+            globalSearch: false,
+
+            /**
+             * Display headers for each group.
+             * @type {boolean}
+             */
+            displayGroupHeader: false,
+
+            /**
+             * function called after component is refreshed.
+             * @type {function}
+             */
+            handleAfterRefresh: function(searchText){}
+        };
+    },
+
+    /** @inheritdoc */
+    propTypes: {
+        listPropertyName: type('string'),
+        data: type('array'),
+        lineComponent: type('func', true),
+        showAll: type('bool'),
+        displayGroupSize: type('bool'),
+        globalSearch: type('bool'),
+        quickSearchInput: type('string'),
+        withQuickSearch: type('bool')
+    },
+
+    /** @inheritdoc */
+    getInitialState(){
+        return this._getStateFromStore();
+    },
+
+    /**
+     * Get component state using store data.
+     * @param {array} data to use.
+     * @returns {object} state.
+     * @private
+     */
+     _getStateFromStore(newData){
+        this._sortData(newData);
```

Comment:
strange, maybe there is a naming confusion, `getStateFromStore` should read data from a store and not from a given JSON.